common/src/main/java/com/microsoft/identity/common/CodeMarker.java

```diff
@@ -0,0 +1,114 @@
+    private String appRecv = null;
+    private String battery = null;
+    private String systemDiskRead = null;
+    private String systemDiskWrite = null;
+
+    public CodeMarker(final String marker, final long timeInMilliseconds, final String timeStamp, final long threadId) {
+        this.marker = marker;
+        this.timeInMilliseconds = timeInMilliseconds;
+        this.timeStamp = timeStamp;
+        this.threadId = threadId;
+    }
+
+    // Returns list of pairs of (key,value) where key is the representation of heading in csv and value is the measurable value.
+    private List<Pair<String,String>> getKeyValuePairsOfCodeMarker() {
+        List<Pair<String,String>> csvKeyValuePairs = new ArrayList<Pair<String,String>>();
+        csvKeyValuePairs.add(new Pair("TimeStamp", this.timeStamp == null ? CodeMarker.csvNoValue : this.timeStamp));
+        csvKeyValuePairs.add(new Pair("Marker", this.marker == null ? CodeMarker.csvNoValue : this.marker));
+        csvKeyValuePairs.add(new Pair("Time", Long.toString(this.timeInMilliseconds)));
+        csvKeyValuePairs.add(new Pair("Thread", Long.toString(this.threadId)));
+        csvKeyValuePairs.add(new Pair("CpuUsed", this.cpuUsed == null ? CodeMarker.csvNoValue : this.cpuUsed));
+        csvKeyValuePairs.add(new Pair("CpuTotal", this.cpuTotal == null ? CodeMarker.csvNoValue : this.cpuTotal));
+        csvKeyValuePairs.add(new Pair("ResidentSize", this.residentSize == null ? CodeMarker.csvNoValue : this.residentSize));
+        csvKeyValuePairs.add(new Pair("VirtualSize", this.virtualSize == null ? CodeMarker.csvNoValue : this.virtualSize));
+        csvKeyValuePairs.add(new Pair("WifiSent", this.wifiSent == null ? CodeMarker.csvNoValue : this.wifiSent));
+        csvKeyValuePairs.add(new Pair("WifiRecv", this.wifiRecv == null ? CodeMarker.csvNoValue : this.wifiRecv));
+        csvKeyValuePairs.add(new Pair("WwanSent", this.wwanSent == null ? CodeMarker.csvNoValue : this.wwanSent));
+        csvKeyValuePairs.add(new Pair("WwanRecv", this.wwanRecv == null ? CodeMarker.csvNoValue : this.wwanRecv));
+        csvKeyValuePairs.add(new Pair("AppSent", this.appSent == null ? CodeMarker.csvNoValue : this.appSent));
+        csvKeyValuePairs.add(new Pair("AppRecv", this.appRecv == null ? CodeMarker.csvNoValue : this.appRecv));
+        csvKeyValuePairs.add(new Pair("Battery", this.battery == null ? CodeMarker.csvNoValue : this.battery));
+        csvKeyValuePairs.add(new Pair("SystemDiskRead", this.systemDiskRead == null ? CodeMarker.csvNoValue : this.systemDiskRead));
+        csvKeyValuePairs.add(new Pair("SystemDiskWrite", this.systemDiskWrite == null ? CodeMarker.csvNoValue : this.systemDiskWrite));
+        return csvKeyValuePairs;
+    }
+
+    public String getCSVHeader() {
+        StringBuilder csvStringBuilder = new StringBuilder();
+        List<Pair<String,String>> csvKeyValuePairs = getKeyValuePairsOfCodeMarker();
+        for(int i = 0; i < csvKeyValuePairs.size(); i++) {
```

Comment:
As a general codestyle (and this is somewhat subjective), I don't favor single letter variables: even for loop indexes. I prefer using `ii` and `jj` as you can `ctrl` + `f` find them in a file, unlike a single letter variable. Just a thought 😉